src/Server.php

```diff
@@ -11,16 +11,19 @@ class Server extends EventEmitter implements ServerInterface
 {
     private $io;
 
-    public function __construct(SocketServerInterface $io)
+    private $maxHeaderSize;
+
+    public function __construct(SocketServerInterface $io, $maxHeaderSize = 4096 /* 4 KiB */)
```

Comment:
Then it might make sense to create `RequestHeaderParserFactory` that gets injected and creates `RequestHeaderParser`s for request (internally it can use cloning). However this might be a little bit of over-engineering.